src/services/squad_spawner/squad_spawner_save_files.cpp

```diff
@@ -0,0 +1,171 @@
+		j["vehinvincibility"]        = s.m_veh_invincibility;
+		j["pedproofs"]["headshot"]   = s.m_ped_proofs[0];
+		j["pedproofs"]["bullet"]     = s.m_ped_proofs[1];
+		j["pedproofs"]["flame"]      = s.m_ped_proofs[2];
+		j["pedproofs"]["melee"]      = s.m_ped_proofs[3];
+		j["pedproofs"]["explosion"]  = s.m_ped_proofs[4];
+		j["pedhealth"]               = s.m_ped_health;
+		j["pedarmor"]                = s.m_ped_armor;
+		j["pedacurracy"]             = s.m_ped_accuracy;
+		j["spawndistance"]           = s.m_spawn_distance;
+		j["squadsize"]               = s.m_squad_size;
+		j["spawndistancemode"]       = s.m_spawn_distance_mode;
+		j["combatabilitylevel"]      = s.m_combat_ability_level;
+		j["stayinveh"]               = s.m_stay_in_veh;
+		j["spawnbehindsamevelocity"] = s.m_spawn_behind_same_velocity;
+		j["disperse"]                = s.m_disperse;
+
+		return j;
+	}
+
+	squad squad_spawner::from_json(nlohmann::json j)
+	{
+		squad new_squad;
+		std::string_view name_sv       = j.value<std::string_view>("name", "");
+		std::string_view desc_sv       = j.value<std::string_view>("description", "");
+		std::string_view ped_model_sv  = j.value<std::string_view>("pedmodel", "");
+		std::string_view veh_model_sv  = j.value<std::string_view>("vehmodel", "");
+		std::string_view weap_model_sv = j.value<std::string_view>("weapmodel", "");
+
+		strcpy(new_squad.m_name, name_sv.data());
+		strcpy(new_squad.m_description, desc_sv.data());
+		strcpy(new_squad.m_ped_model, ped_model_sv.data());
+		strcpy(new_squad.m_vehicle_model, veh_model_sv.data());
+		strcpy(new_squad.m_weapon_model, weap_model_sv.data());
+
+		new_squad.m_ped_invincibility = j.value("pedinvincibility", 0);
+		new_squad.m_veh_invincibility = j.value("vehinvincibility", 0);
+
+		if (j.contains("pedproofs"))
+		{
+			new_squad.m_ped_proofs[0] = j["pedproofs"].value("headshot", false);
+			new_squad.m_ped_proofs[1] = j["pedproofs"].value("bullet", false);
+			new_squad.m_ped_proofs[2] = j["pedproofs"].value("flame", false);
+			new_squad.m_ped_proofs[3] = j["pedproofs"].value("melee", false);
+			new_squad.m_ped_proofs[4] = j["pedproofs"].value("explosion", false);
+		}
+		new_squad.m_ped_health                 = j.value("pedhealth", 100);
+		new_squad.m_ped_armor                  = j.value("pedarmor", 0);
+		new_squad.m_ped_accuracy               = j.value("pedacurracy", 50);
+		new_squad.m_spawn_distance             = j.value("spawndistance", 0);
+		new_squad.m_squad_size                 = j.value("squadsize", 1);
+		new_squad.m_spawn_distance_mode        = (eSquadSpawnDistance)j.value("spawndistancemode", 1);
+		new_squad.m_combat_ability_level       = (eCombatAbilityLevel)j.value("combatabilitylevel", 2);
+		new_squad.m_stay_in_veh                = j.value("stayinveh", false);
+		new_squad.m_spawn_behind_same_velocity = j.value("spawnbehindsamevelocity", true);
+		new_squad.m_disperse                   = j.value("disperse", false);
+
+		return new_squad;
+	}
```

Comment:
Replace this code with `NLOHMANN_DEFINE_TYPE_INTRUSIVE`.

You can then easily parse the json to the struct and go from the struct to json.
```cpp
// go from squad struct to JSON
nlohmann::json j = sqd;

// go from JSON to squad struct
auto sqd = j.get<squad>();
```

If you want to get an idea of how it works take a look at the `src/services/vehicle/handling_profile.hpp` file as well as the `src/services/vehicle/handling_service.cpp` file.

Reply:
Understood